tools/lambda-promtail/lambda-promtail/main.go

```diff
@@ -22,15 +22,21 @@ const (
 	maxErrMsgLen = 1024
 )
 
+type ExtraLabel struct {
+	key   string
+	value string
+}
+
 var (
-	writeAddress       *url.URL
-	username, password string
-	keepStream         bool
-	batchSize          int
-	s3Clients          map[string]*s3.Client
+	writeAddress                       *url.URL
+	username, password, extraLabelsRaw string
+	keepStream                         bool
+	batchSize                          int
+	s3Clients                          map[string]*s3.Client
+	extraLabels                        []ExtraLabel
 )
 
-func init() {
+func configure() {
```

Comment:
I am not super familiar with Go, but having this as `init()` was making it impossible to run the tests. I renamed to `configure()` and call it from `main()` let me know if that is a problem.

Reply:
Lets name this something a bit more descriptive like `setupArguments`. Here's a description of what the init function is: https://go.dev/doc/effective_go#init